File: backend/app/models/asset.py
"""Pydantic models for assets."""
from pydantic import BaseModel, Field, HttpUrl
from typing import Optional, List
from datetime import datetime
from uuid import UUID


class AssetCreate(BaseModel):
    """Model for creating new asset."""
    file_name: str
    file_size: int = Field(gt=0)
    width: int = Field(gt=0)
    height: int = Field(gt=0)
    generation_prompt: Optional[str] = None
    generation_model: str
    style_tags: List[str] = []
    project_name: Optional[str] = None


class AssetResponse(BaseModel):
    """Model for asset API responses."""
    id: str
    user_id: str
    file_url: str
    file_name: str
    file_size: int
    mime_type: str
    width: int
    height: int
    generation_prompt: Optional[str]
    generation_model: str
    style_tags: List[str]
    project_name: Optional[str]
    is_favorite: bool
    metadata: dict
    created_at: str
    updated_at: str


class AssetFilter(BaseModel):
    """Model for filtering assets."""
    project_name: Optional[str] = None
    style_tags: Optional[List[str]] = None
    search_query: Optional[str] = None
    limit: int = Field(50, le=100)
    offset: int = 0


class AssetsListResponse(BaseModel):
    """Response for listing assets."""
    assets: List[AssetResponse]
    total: int
    limit: int
    offset: int


class AssetDeleteResponse(BaseModel):
    """Response for asset deletion."""
    success: bool
    message: str

File: frontend/src/types/asset.ts
export interface Asset {
  id: string;
  user_id: string;
  file_url: string;
  file_name: string;
  file_size: number;
  mime_type: string;
  width: number;
  height: number;
  generation_prompt: string | null;
  generation_model: string;
  style_tags: string[];
  project_name: string | null;
  is_favorite: boolean;
  metadata: Record<string, any>;
  created_at: string;
  updated_at: string;
  // Phase 2: Versioning fields
  parent_asset_id: string | null;
  version_number: number;
  refinement_instruction: string | null;
}

export interface AssetsListResponse {
  assets: Asset[];
  total: number;
  limit: number;
  offset: number;
}
